kotest-bom/build.gradle.kts

```diff
@@ -0,0 +1,33 @@
+plugins {
+   `java-platform`
+   `maven-publish`
+}
+
+version = Ci.publishVersion
+
+val bomProject = project
+val excludeFromBom = listOf("kotest-examples", "kotest-tests")
+fun projectsFilter(candidateProject: Project) =
+   excludeFromBom.all { !candidateProject.name.contains(it) }
+      && candidateProject.name != bomProject.name
+
+rootProject.subprojects.filter(::projectsFilter).forEach { bomProject.evaluationDependsOn(it.path) }
```

Comment:
What does evaluationDependsOn do, can you add a comment ?

Reply:
I've added a comment but I wanted to say something here as well. We don't want to end up with non-publishing subprojects contributing to the BOM (it won't really matter, as a BOM just says what version to use if you do end up using a given dependency) so we filter out those.

`evalutationDependsOn(":subproject:)` declares that the current project has a dependency on  `:subproject` and configuration of the current project triggers configuration of `:subproject`.

The reason this is necessary is that we use the `constraints` block to declare a dependency constraint and not a hard dependency on the project.

I'm not sure how well I've explained it actually. Is that reasonably clear?

Reply:
I don't really understand the bit about constraints and why you did that.